### PR TITLE
Check for positive amounts

### PR DESCRIPTION
**Pull request by ggobugi27**
- Check that `mint`/`burn`/`sale` amount is positive 
- Check that `min-amount`, `max-supply` is positive
- For fixed-quote-royalty-policy, add conditionals to case where `royalty-rate` is 0.0, so that transfer with 0.0 is prevented and enforce-unit checks `sale-price`

### CODE REVIEW

**Review by buckie — APPROVED** [review]

File: pact/fixed-quote-royalty-policy.pact [hunk 1]
```diff
@@ -124,8 +128,11 @@
             (recipient-guard:guard (at 'recipient-guard spec))
             (recipient-details:object (fungible::details recipient))
             (sale-price:decimal (* amount price)) )
-      (fungible::enforce-unit (* sale-price royalty-rate))
-      (enforce (< 0.0 price) "Offer amount must be positive")
+      (if
+        (= royalty-rate 0.0)
+        (fungible::enforce-unit sale-price)
```
Comment (sirlensalot) [inline pact/fixed-quote-royalty-policy.pact:133]: This needs to not be conditional; it's the following test that needs the condition

File: pact/fixed-quote-royalty-policy.pact [hunk 2]
```diff
@@ -87,10 +89,12 @@
             (royalty-rate:decimal (at 'royalty-rate spec))
             (creator-details:object (fungible::details creator ))
             )
-      (fungible::enforce-unit royalty-rate)
+      (enforce (>= min-amount 0.0) "Invalid min-amount")
+      (enforce (>= max-supply 0.0) "Invalid max-supply")
       (enforce (=
         (at 'guard creator-details) creator-guard)
         "Creator guard does not match")
+      (fungible::enforce-unit royalty-rate)
```
Comment (sirlensalot) [inline pact/fixed-quote-royalty-policy.pact:97]: I don't think this is necessary ... in any case it won't stop precision problems below.

File: pact/fixed-quote-royalty-policy.pact [hunk 3]
```diff
@@ -154,7 +161,10 @@
           , 'recipient-guard := recipient-guard:guard
           }
           (let ((sale-price (* amount price)))
-            (fungible::transfer-create buyer creator creator-guard (* sale-price royalty-rate))
+            (if
+              (> (* sale-price royalty-rate) 0.0)
+              (fungible::transfer-create buyer creator creator-guard (* sale-price royalty-rate))
+              true)
```
Comment (sirlensalot) [inline pact/fixed-quote-royalty-policy.pact:167]:
```suggestion
              "No royalty")
```

Comment (sirlensalot) [reply pact/fixed-quote-royalty-policy.pact:167]: We should `typecheck` this module (and the other policies) in the repl and fix any type errors

**Review by sirlensalot — APPROVED** [review]

File: pact/fixed-quote-royalty-policy.pact (hunk 1 above)
Comment (ggobugi27) [reply pact/fixed-quote-royalty-policy.pact]: Removed the conditionals, and floored the `royalty-payout`

**Review by sirlensalot — APPROVED** [review]

**Pull request merged**